pkg/provisioner/ironic/testserver/server.go

```diff
@@ -223,7 +221,10 @@ func (m *MockServer) defaultHandler(w http.ResponseWriter, r *http.Request) {
 	method := r.Method
 
 	for _, response := range m.defaultResponses {
-		if response.method == "" || response.method == method {
+		// gocritic produces the following error here which can be overlooked
+		// nestingReduce: invert if cond, replace body with `continue`,
+		// move old body after the statement (gocritic)
```

Comment:
As commented, linter output was unclear, IMO this can be overlooked

Reply:
It would like you to do
```
if response.method != "" && response.method != method {
  continue
}
// the code in the current if block here, ie. invert the if condition
```
to reduce the nesting level of code by 1, since you don't have else case.

Reply:
how I understood is the `continue` in line 230 needs to be the `continue` here but the loop should only continue when `match` is nill.

Reply:
I think you are right, I am just overthinking here, I will test it

Reply:
Works! Thanks.